app/scripts/models/user.js

```diff
@@ -350,6 +358,44 @@ define(function (require, exports, module) {
         });
     },
 
+    /**
+     * Complete signup for the account. Notifies other tabs of signin
+     * if the account has a sessionToken and verification successfully
+     * completes.
+     *
+     * @param {object} account - account to verify
+     * @param {string} code - verification code
+     * @returns {promise} - resolves with the account when complete
+     */
+    completeAccountSignUp: function (account, code) {
+      var self = this;
+
+      // The original tab may no longer be open to notify other
+      // windows the user is signed in. If the account has a `sessionToken`,
+      // the user verified in the same browser. Notify any tabs that care.
+      function notifyIfSignedIn(account) {
+        if (account.has('sessionToken')) {
+          self._notifyOfAccountSignIn(account);
+        }
+      }
+
+      return account.verifySignUp(code)
+        .fail(function (err) {
+          if (MarketingEmailErrors.created(err)) {
+            // A basket error should not prevent the
+            // sign up verification from completing, Notify of
+            // sign in and continue.
+            notifyIfSignedIn(account);
```

Comment:
Maybe I should change it to:

> A MarketingEmailError doesn't prevent a user from completing the signup. If we receive a MarketingEmailError, notify other tabs of the sign in, and re-throw the error so it can be logged at a higher level.

Any clearer?